### PR TITLE
Strona logowania

### LINKED ISSUE #8: Login page
W tym issue należy wprowadzić również koncepcję routingu w aplikacji i jako domyślną stronę (przy braku kluczy w local storage) wyświetlić login page.

### PR DESCRIPTION
**Pull request by SiwyDym**
resolves #8

### CODE REVIEW

File: src/main/components/LoginPage.vue [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      <v-flex xs12>
+        <v-text-field
+          id="password"
+          label="Password"
+          type='password'
+          v-model="loginPassword"
+          @keyup.enter="submit($event)"
+        ></v-text-field>
+      </v-flex>
+      <input type="checkbox" id="rememberMe" v-model="rememberMe"/>
```
Comment (kacpak) [inline src/main/components/LoginPage.vue]: Są do tego komponenty w Vuetify - dlaczego sam próbowałeś je zrobić?

File: src/main/components/LoginPage.vue [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+<template>
+  <div class="user-modal-container" id="login-modal">
+    <div class="form-login" id="form-login">
+      <h5>Please login:</h5>
+      <v-flex xs12>
+        <v-text-field
+          id="user"
+          label="Name"
+          v-model="loginUser"
+          @keyup.enter="submit($event)"
```
Comment (kacpak) [inline src/main/components/LoginPage.vue]: Wydaje mi się że można się podpiąć pod submit w całym formularzu?
Czemu to w ogóle nie jest jako form?

**Comment by kacpak** [comment]
Ogólnie prośba, wklej tutaj screen jak wygląda interfejs po zmianach zawsze, będzie szybciej i łatwiej o response od designu np

**Comment by kacpak** [comment]
localhost_8080_

File: src/main/components/LoginPage.vue (hunk 2 above)
Comment (SiwyDym) [reply src/main/components/LoginPage.vue]: Zamienie na form

File: src/main/components/LoginPage.vue (hunk 1 above)
Comment (SiwyDym) [reply src/main/components/LoginPage.vue]: Zamienie na component z Vuetify

**Comment by kacpak** [comment]
Sprawdź najpierw drugi pull request

wt., 2.01.2018, 13:09 użytkownik SiwyDym <notifications@github.com> napisał:

**Comment by SiwyDym** [comment]
Mateusz Twój commit zawiera błędy i warningi, możesz poprawic? 

> C:\Users\emarsiw\Downloads\ideaIU-2017.1.win\bin\runnerw.exe "F:\Programs files\NodeJS\node.exe" "F:\Programs files\NodeJS\node_modules\npm\bin\npm-cli.js" run dev:web --scripts-prepend-node-path=auto

> chat-app@0.0.0 dev:web F:\Projekty\app
> webpack-dev-server --config .build-scripts/webpack.web.config.js

Project is running at https://localhost:8080/
webpack output is served from /
404s will fallback to /index.html
Broadcasting "http" with subtype of "webpack" via ZeroConf DNS (Bonjour)
webpack: wait until bundle finished: /
Hash: 39b6f5c6d5d029b799b3
Version: webpack 3.8.1
Time: 7457ms
 12 assets
   [0] ./node_modules/moment/moment.js 130 kB {4} [built]
   [5] ./node_modules/vue/dist/vue.esm.js 282 kB {4} [built]
  [23] (webpack)/hot/log.js 1 kB {4} [built]
 [184] multi (webpack)-dev-server/client?https://localhost:8080 webpack/hot/dev-server ./src/main/main.js 52 bytes {4} [built]
 [185] (webpack)-dev-server/client?https://localhost:8080 7.95 kB {4} [built]
 [186] ./node_modules/url/url.js 23.3 kB {4} [built]
 [192] ./node_modules/strip-ansi/index.js 161 bytes {4} [built]
 [194] ./node_modules/loglevel/lib/loglevel.js 7.74 kB {4} [built]
 [195] (webpack)-dev-server/client/socket.js 1.05 kB {4} [built]
 [197] (webpack)-dev-server/client/overlay.js 3.73 kB {4} [built]
 [202] (webpack)/hot nonrecursive ^\.\/log$ 170 bytes {4} [built]
 [204] (webpack)/hot/dev-server.js 1.56 kB {4} [built]
 [205] (webpack)/hot/log-apply-result.js 1.27 kB {4} [built]
 [206] ./src/main/main.js 645 bytes {4} [built]
 [209] ./node_modules/axios/index.js 40 bytes {4} [built]
    + 302 hidden modules

WARNING in ./src/main/services/auth.service.js

  ⚠  http://eslint.org/docs/rules/no-console  Unexpected console statement  
  src\main\services\auth.service.js:18:3
    console.log(username, password);
     ^


✘ 1 problem (0 errors, 1 warning)


Warnings:
  1  http://eslint.org/docs/rules/no-console
 @ ./src/main/store/modules/Auth.js 3:0-61
 @ ./src/main/store/modules nonrecursive \.js$
 @ ./src/main/store/modules/index.js
 @ ./src/main/store/index.js
 @ ./src/main/main.js
 @ multi (webpack)-dev-server/client?https://localhost:8080 webpack/hot/dev-server ./src/main/main.js

WARNING in ./src/main/store/modules/Auth.js

  ⚠  http://eslint.org/docs/rules/no-console  Unexpected console statement  
  src\main\store\modules\Auth.js:27:7
        console.error(e);
         ^


✘ 1 problem (0 errors, 1 warning)


Warnings:
  1  http://eslint.org/docs/rules/no-console
 @ ./src/main/store/modules nonrecursive \.js$
 @ ./src/main/store/modules/index.js
 @ ./src/main/store/index.js
 @ ./src/main/main.js
 @ multi (webpack)-dev-server/client?https://localhost:8080 webpack/hot/dev-server ./src/main/main.js

ERROR in ./src/main/App.vue
Module not found: Error: Can't resolve 'stylus-loader' in 'F:\Projekty\app\src\main'
 @ ./src/main/App.vue 4:2-380
 @ ./src/main/main.js
 @ multi (webpack)-dev-server/client?https://localhost:8080 webpack/hot/dev-server ./src/main/main.js
Child html-webpack-plugin for "index.html":
     1 asset
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/main/index.ejs 777 bytes {0} [built]
       [1] ./node_modules/lodash/lodash.js 540 kB {0} [built]
       [2] (webpack)/buildin/global.js 488 bytes {0} [built]
       [3] (webpack)/buildin/module.js 495 bytes {0} [built]
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/mdi/css/materialdesignicons.css:
     5 assets
       [0] ./node_modules/css-loader!./node_modules/mdi/css/materialdesignicons.css 124 kB {0} [built]
       [1] ./node_modules/css-loader/lib/css-base.js 2.26 kB {0} [built]
       [2] ./node_modules/mdi/fonts/materialdesignicons-webfont.eot?v=2.0.46 116 bytes {0} [built]
       [3] ./node_modules/mdi/fonts/materialdesignicons-webfont.eot 116 bytes {0} [built]
       [4] ./node_modules/mdi/fonts/materialdesignicons-webfont.woff2?v=2.0.46 118 bytes {0} [built]
       [5] ./node_modules/mdi/fonts/materialdesignicons-webfont.woff?v=2.0.46 117 bytes {0} [built]
       [6] ./node_modules/mdi/fonts/materialdesignicons-webfont.ttf?v=2.0.46 116 bytes {0} [built]
       [7] ./node_modules/mdi/fonts/materialdesignicons-webfont.svg?v=2.0.46 116 bytes {0} [built]

**Comment by kacpak** [comment]
Mój kod zawiera tylko warningi, które nie przeszkadzają w buildzie (te console logi są do testów z tego co pamiętam).
A błąd kompilacji jest po Twojej stronie - doszły nowe zależności, więc musisz je dociągnąć.
W głównym katalogu projektu odpal `yarn`.

**Comment by SiwyDym** [comment]
Nie wyświetlał się material icons dla vuetify, bo nie było dodane [https://vuetifyjs.com/vuetify/quick-start]( vuetify css) do index.ejs. 

Login page poprawiony do oficjalnej już wersji skórki.

**Comment by SiwyDym** [comment]
image

File: src/main/containers/guest/Login.vue [hunk 3]
```diff
@@ -1,17 +1,87 @@
...
+    data: () => ({
+      valid: true,
+      username: '',
+      password: '',
+      select: null,
+      checkbox: false
+    }),
+    methods: {
+      async login() {
+        await this.$store.dispatch('authenticate', { username: 'test', password: '1234' }); // Użyj danych z formularza
```
Comment (kacpak) [inline src/main/containers/guest/Login.vue]: przekaż tu dane z formularza i zależnie od wyniku tej akcji (user istnieje i ma dobre hasło) to przenosi go na następną stronę (linijka niżej)

File: src/main/index.ejs [hunk 4]
```diff
@@ -4,6 +4,10 @@
     <meta charset="utf-8">
     <meta http-equiv="x-ua-compatible" content="ie=edge">
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+    <link href='https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Material+Icons' rel="stylesheet">
+    <link href="https://unpkg.com/vuetify/dist/vuetify.min.css" rel="stylesheet">
```
Comment (kacpak) [inline src/main/index.ejs]: ten css już jest zincludowany przez .styl w App.vue, dlaczego jest wymagany raz jeszcze?
czcionka może zostać, choć też powinna być dodana w procesie budowania

File: src/main/index.ejs [hunk 5]
```diff
@@ -4,6 +4,10 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1, user-scalable=no, minimal-ui">
```
Comment (kacpak) [inline src/main/index.ejs]: meta viewport istnieje już kilka linijek wyżej

**Comment by kacpak** [comment]
Daj screeny dla różnych szerokości ekranu to zobaczymy jak się skaluje na telefonach

File: src/main/index.ejs (hunk 4 above)
Comment (SiwyDym) [reply src/main/index.ejs]: To coś źle zostało to dodane, powinno być w index.html, tak w każdym jest napisane w dockach. Mimo wszytsko jak usuniesz css'a, to mateiral icons z vuetify nie działają.

File: src/main/containers/guest/Login.vue (hunk 3 above)
Comment (SiwyDym) [reply src/main/containers/guest/Login.vue]: ` async login() {
        const ok = await this.$store.dispatch('authenticate', { username: this.username, password: this.password });

        if (ok) {
          return this.$router.push({ name: 'main' });
        }
        return null;
      },`
Coś takiego? Nie wiem czy dobrze Cie zrozumaiłem.

File: src/main/index.ejs (hunk 4 above)
Comment (SiwyDym) [reply src/main/index.ejs]: @kacpak  ?

Comment (kacpak) [reply src/main/index.ejs]: to wywal viewport, a ja później sfixuje fonty tak jak być powinno (nie tutaj) - dokumentacja daje Ci tylko najprostszą ścieżkę, my używamy zmodyfikowanych css'ów by mieć inne kolory bez nadpisywania w każdym miejscu

**Comment by kacpak** [comment]
Następnym razem nie edytuj mojego komentarza tylko wstaw swój nowy :P
Screen dla mobile ma za krótkie pola moim zdaniem, popraw to i będzie ok

File: src/main/index.ejs (hunk 4 above)
Comment (SiwyDym) [reply src/main/index.ejs]: Spoczko, tylko nie działa na razie wyswietlanie cssvuetify :|  Usune fonty i viewport w takim razie. 

Daj zanć co do komentarza o metodzie `login`.

Comment (kacpak) [reply src/main/index.ejs]: Co nie działa? Te fonty nawet jak tutaj są?

File: src/main/containers/guest/Login.vue (hunk 3 above)
Comment (kacpak) [reply src/main/containers/guest/Login.vue]: Może być (nie pamiętam co zwraca authenticate), mniej więcej o coś takiego chodzi.
`return null` jest nie potrzebny

Z tym że w przypadku błędu powinniśmy dostać jakąś informację na ekranie że coś poszło nie tak, zobacz jak to się robi w formularzach vuetify by było ładnie i napisz że nie udało się zalogować lub coś w tym stylu

File: src/main/index.ejs (hunk 4 above)
Comment (SiwyDym) [reply src/main/index.ejs]: Nie wyświetla się ikonka checkboxa, tylko surowy text.

Comment (kacpak) [reply src/main/index.ejs]: W ogóle? nawet z Twoimi zmianami?

Comment (kacpak) [reply src/main/index.ejs]: Moim zdaniem funkcjonalność 'zapamiętaj mnie' i tak możemy na razie olać, więc możesz ją zakomentować w templat'cie

Comment (SiwyDym) [reply src/main/index.ejs]: Ze zmianami działa.

**Comment by kacpak** [comment]
To na razie wywal tylko viewport, później sam poprawię by ikony działały
bez tego

czw., 4 sty 2018 o 15:03 użytkownik SiwyDym <notifications@github.com>
napisał:

File: src/main/containers/guest/Login.vue (hunk 3 above)
Comment (SiwyDym) [reply src/main/containers/guest/Login.vue]: Eslint jest innego zdania: 
`  ✘  http://eslint.org/docs/rules/consistent-return  Async method 'login' expected a return value  `

**Comment by kacpak** [comment]
Hah
No to niech zostanie :-)

czw., 4.01.2018, 15:08 użytkownik SiwyDym <notifications@github.com>
napisał:

**Comment by SiwyDym** [comment]
image
image
image

File: src/main/containers/guest/Login.vue [hunk 6]
```diff
@@ -1,17 +1,104 @@
...
+        if (isUser) {
+          return this.$router.push({ name: 'main' });
+        }
+        return null;
+      },
+      signup() {
+        return this.$router.push({ name: 'register' });
+      },
+      onDismissed() {
+        this.$store.commit('clearError');
```
Comment (kacpak) [inline src/main/containers/guest/Login.vue]: czemu to jest w store zamiast lokalnie? globalnie apliacja tego nie potrzebuje

Comment (SiwyDym) [reply src/main/containers/guest/Login.vue]: Tak jak rozmawialiśmy, ostatecznie zostaje, że error jest w store.

**Pull request merged**